vivisect/qt/views.py

```diff
@@ -238,15 +287,89 @@ def vqLoad(self):
+
+        self.menu = QMenu(self)
+        self.m_caseSensitivityAction = self.menu.addAction("Case Sensitive")
+        self.m_caseSensitivityAction.setCheckable(True)
+        self.m_caseSensitivityAction.toggled.connect(self.filterChanged)
+
+
+        self.menu.addSeparator();
+        self.m_patternGroup.setExclusive(True);
+
+        self.patternAction = self.menu.addAction("Fixed String");
+        self.patternAction.setData(QtCore.QVariant(int(QtCore.QRegExp.FixedString)));
+        self.patternAction.setCheckable(True);
+        self.patternAction.setChecked(True);
+        self.m_patternGroup.addAction(self.patternAction);
+
+        self.patternAction = self.menu.addAction("Regular Expression");
+        self.patternAction.setCheckable(True);
+        self.patternAction.setData(QtCore.QVariant(int(QtCore.QRegExp.RegExp2)));
+        self.m_patternGroup.addAction(self.patternAction);
+
+        self.patternAction = self.menu.addAction("Wildcard");
+        self.patternAction.setCheckable(True);
+        self.patternAction.setData(QtCore.QVariant(int(QtCore.QRegExp.Wildcard)));
+        self.m_patternGroup.addAction(self.patternAction);
+        
+        self.m_patternGroup.triggered.connect(self.filterChanged);
+
+        self.icon = QtGui.QIcon(QtGui.QPixmap(":/images/find.png"))
+        self.optionsButton = QToolButton()
+        self.optionsButton.setCursor(QtCore.Qt.ArrowCursor)
+        self.optionsButton.setFocusPolicy(QtCore.Qt.NoFocus)
+        self.optionsButton.setStyleSheet("* { border: none; }")
+        self.optionsButton.setIcon(self.icon)
+        self.optionsButton.setMenu(self.menu)
+        self.optionsButton.setPopupMode(QToolButton.InstantPopup)
+
+        self.optionsAction = QWidgetAction(self)
+        self.optionsAction.setDefaultWidget(self.optionsButton)
+        self.addAction(self.optionsAction, QLineEdit.LeadingPosition)
+
+    def caseSensitivity(self):
+        return  (QtCore.Qt.CaseInsensitive, QtCore.Qt.CaseSensitive)[self.m_caseSensitivityAction.isChecked()]
+
+    def setCaseSensitivity(self, cs):
+        self.m_caseSensitivityAction.setChecked(cs == QtCore.Qt.CaseSensitive)
+
+    def patternSyntax(self):
+        return self.patternSyntaxFromAction(self.m_patternGroup.checkedAction())
+
+    def setPatternSyntax(self, s):
+        for a in self.m_patternGroup.actions():
+            if (self.patternSyntaxFromAction(a) == s):
+                a.setChecked(True)
+                break
+
+    def patternSyntaxFromAction(self, a):
+        return int(a.data())
+
+class VQVivFunctionsViewPart(VQVivTreeView):
 
     _viv_navcol = 0
     window_title = 'Functions'
     columns = ('Name','Address', 'Size', 'Ref Count')
 
     def __init__(self, vw, vwqgui):
-        VQVivTreeView.__init__(self, vw, vwqgui)
-        self.setModel( VivNavModel(self._viv_navcol, self, columns=self.columns) )
+        VQVivTreeView.__init__(self, vw, vwqgui, withfilter=True)
```

Comment:
Tried to open the function listing
```
Traceback (most recent call last):
  File "/home/james/work/vivisect/vqt/menubuilder.py", line 87, in __call__
    return self.callback(*self.args, **self.kwargs)
  File "/home/james/work/vivisect/vivisect/qt/main.py", line 396, in _menuViewFunctions
    self.vqBuildDockWidget('VQVivFunctionsView', area=QtCore.Qt.RightDockWidgetArea)
  File "/home/james/work/vivisect/vqt/application.py", line 130, in vqBuildDockWidget
    obj = cls(*args)
  File "/home/james/work/vivisect/vivisect/qt/views.py", line 166, in __init__
    self.view = self.view_type(vw, vwqgui, *args, **kwargs)
  File "/home/james/work/vivisect/vivisect/qt/views.py", line 365, in __init__
    VQVivTreeView.__init__(self, vw, vwqgui, withfilter=True)
  File "/home/james/work/vivisect/vivisect/qt/views.py", line 85, in __init__
    vq_tree.VQTreeView.__init__(self, parent=vwqgui, **kwargs)
  File "/home/james/work/vivisect/vqt/tree.py", line 218, in __init__
    QTreeView.__init__(self, parent=parent, **kwargs)
TypeError: 'withfilter' is an unknown keyword argument
```
Might need to pop the withfilter kwarg in VQVivTreeView.